src/service/url-expander/expander.js

```diff
@@ -168,20 +172,23 @@ export class Expander {
           urlIndex = match.stop + 1;
           match = matches[++matchIndex];
 
+          // Collect any chars that may be prefixing the macro, if we are in
+          // a nested context trim the args.
+          if (builder.trim().length) {
```

Comment:
nit: Since we parse the string from left to right could we append the builder to result first before we handle with the matched binding.

Reply:
Also I understand this is used for 
```
CLIENT_ID(   SUBSTR(abc, 1, 2))
```
or 
```
CLIENT_ID(  testSUBSTR(abc, 1, 2))
```
However it also applies to
```
CLIENT_ID (   test  SUBSTR(abc, 1, 2))
```
What do you think should be the different between these cases?

Reply:
for the first comment, sure we can move it up before finding the binding. For the second comment you are right these will all behave the same. The hard part is the distinction between `CLIENT_ID (   test  SUBSTR(abc, 1, 2))` and `CLIENT_ID (a,  test  SUBSTR(abc, 1, 2))`. We could only trim the first space instead of trimming the whole thing. What do you think?

Reply:
Hmmm `CLIENT_ID ( test SUBSTR(abc, 1, 2))` maybe a bad example, because there's a space between `CLIENT_ID` and `(`. I don't think we can handle that. Let me know if that's true and expected. 

`CLIENT_ID(a,  test  SUBSTR(abc, 1, 2))`. So I assume the `  test  ` will be trimmed to `test`. And it equals to `CLIENT_ID(a,testSUBSTR(abc, 1, 2))`. Is that right? 
Do you think we should only remove the trialing space in such case?

Reply:
you are correct these would be equivalent. Do you mean the leading space? If so I think that may be a good idea.

Reply:
just so we are on the same page, I am thinking something like `trimLeft` instead of trim.

Reply:
Ah, you're right. leading space it is : )

Reply:
👍changed to trimStart and moved the builder push before the binding stuff.

Reply:
also added test with the space. ```$IF(QUERY_PARAM(test), foo TIMESTAMP, ``)```